src/pages/feed/FeedScreen.tsx

```diff
@@ -0,0 +1,214 @@
+import { MarkType } from "./FeedNotificationListCardMark";
+
+enum FeedFilter {
+  ALL,
+  NOT_SEEN,
+}
+
+export const FeedScreen = ({ navigation }) => {
+  const { t } = useTranslation();
+
+  const FeedNotificationTypeToMessageTemplateMap = new Map<
+    FeedNotificationType,
+    string
+  >([
+    [
+      FeedNotificationType.EVENT_CREATE,
+      t("feed.message-templates.event-create"),
+    ],
+    [
+      FeedNotificationType.EVENT_UPDATE,
+      t("feed.message-templates.event-update"),
+    ],
+    [
+      FeedNotificationType.EVENT_CANCEL,
+      t("feed.message-templates.event-cancel"),
+    ],
+    [
+      FeedNotificationType.EVENT_REENABLE,
+      t("feed.message-templates.event-reenable"),
+    ],
+    [
+      FeedNotificationType.ORGANIZATION_CREATE,
+      t("feed.message-templates.organization-create"),
+    ],
+    [
+      FeedNotificationType.ORGANIZATION_UPDATE,
+      t("feed.message-templates.organization-update"),
+    ],
+  ]);
+
+  const FeedNotificationTypeToMarkTypeMap = new Map<
+    FeedNotificationType,
+    MarkType
+  >([
+    [FeedNotificationType.EVENT_CREATE, MarkType.CREATE],
+    [FeedNotificationType.EVENT_UPDATE, MarkType.EDIT],
+    [FeedNotificationType.EVENT_CANCEL, MarkType.CANCEL],
+    [FeedNotificationType.EVENT_REENABLE, MarkType.REENABLE],
+    [FeedNotificationType.ORGANIZATION_CREATE, MarkType.CREATE],
+    [FeedNotificationType.ORGANIZATION_UPDATE, MarkType.EDIT],
+  ]);
+
+  const [notifications, setNotifications] = useState<FeedNotification[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [feedFilter, setFeedFilter] = useState<FeedFilter>(FeedFilter.ALL);
+
+  const isFocused = useIsFocused();
+  useEffect(() => {
+    const fetchOrganizationsData = async () => {
+      setIsLoading(true);
+      try {
+        const feedNotifications: FeedNotification[] = await getFeed(
+          feedFilter === FeedFilter.NOT_SEEN,
+        );
+        setNotifications(feedNotifications);
+      } catch (error) {
+        console.log("Fetching feed error", error);
+      }
+      setIsLoading(false);
+    };
+    isFocused && fetchOrganizationsData();
+  }, [feedFilter, isFocused]);
+
+  const handleCardPress = (notification: FeedNotification) => {
+    console.log(`Clicked card: ${notification.type}`);
+
+    if (!notification.seen) {
+      markNotificationAsSeen(notification.id)
+        .then(() => {
+          console.log("Successfully marked notification as seen");
+        })
+        .catch(() => {
+          console.log("Error while marking notification as seen");
+        });
+    }
+
+    if (notification.regardingEventDTO) {
+      navigation.navigate("Event", {
+        eventId: notification.regardingEventDTO.id,
+      });
+    } else if (notification.regardingOrganizationDto) {
+      navigation.navigate("Organization", {
+        organizationId: notification.regardingOrganizationDto.id,
+      });
+    }
+  };
+
+  const renderFeedNotificationItem = ({
```

Comment:
jesli to juz jest takie duze to wrzucmy to w oddzielny komponent FeedNotificationItem